packages/platform-browser/src/internal/runtime.ts

```diff
@@ -2,14 +2,28 @@ import type { RunMain } from "@effect/platform/Runtime"
 import * as Cause from "effect/Cause"
 import * as Effect from "effect/Effect"
 import * as Logger from "effect/Logger"
+import * as Layer from "effect/Layer"
+import * as HashSet from "effect/HashSet"
+import * as FiberRef from "effect/FiberRef"
+
+/** @internal */
+const useStructuredLogger = Layer.fiberRefLocallyScopedWith(FiberRef.currentLoggers, (loggers) =>  {
+  if(HashSet.has(loggers, Logger.defaultLogger)) {
+    const set = HashSet.remove(loggers, Logger.defaultLogger);
+    return HashSet.add(set, Logger.structuredLogger)
+  }
+
+  return loggers;
+})
+
 
 /** @internal */
 export const runMain: RunMain = (
   effect,
   options
 ) => {
-  const _effect = Effect.provide(effect, Logger.structured)
-
+  const _effect = Effect.provide(effect, useStructuredLogger)
```

Comment:
I think instead of using a Layer, we should just use `Effect.locallyWith`

Reply:
Updated

Reply:
I'll hand it over to eslint for the rest of the review :)